src/Tools/dotnet-monitor/GenerateApiKeyCommandHandler.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.IO;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    /// <summary>
+    /// Used to generate Api Key for authentication. The first output is
+    /// part of the Authorization header, and is the Base64 encoded key.
+    /// The second output is a hex encoded string of the hash of the secret.
+    /// </summary>
+    internal sealed class GenerateApiKeyCommandHandler
+    {
+        public Task<int> GenerateApiKey(CancellationToken token, IConsole console)
+        {
+            using RandomNumberGenerator rng = RandomNumberGenerator.Create();
+            using HashAlgorithm hashAlgorithm = SHA256.Create();
+
+            byte[] secret = new byte[32];
+            rng.GetBytes(secret);
+
+            byte[] hash = hashAlgorithm.ComputeHash(secret);
+
+            console.Out.WriteLine(Convert.ToBase64String(secret));
```

Comment:
Can you label these in the console output? It might not be obvious which value is the key and which is the hash.